solution/app/src/main/java/com/google/codelabs/maps/placesdemo/CurrentPlaceActivity.kt

```diff
@@ -167,9 +181,29 @@ class CurrentPlaceActivity : AppCompatActivity() {
         }
     }
 
+    override fun onMapReady(map: GoogleMap) {
+        this.map = map
+    }
+
+    private fun setPlaceOnMap(place: Place?) {
+        val latLng = place?.latLng ?: defaultLocation
+        map?.moveCamera(
+            CameraUpdateFactory.newLatLngZoom(
+                latLng,
+                DEFAULT_ZOOM
+            )
+        )
+        map?.addMarker(
+            MarkerOptions()
+            .position(latLng)
+            .title(place?.name)
+        )
```

Comment:
Let's also include the code from [Save the maps' state](https://developers.google.com/maps/documentation/android-sdk/current-place-tutorial#save-the-maps-state) so we don't lose that content in the migration to this codelab.